tree/treeplayer/src/TTreeProcessorMT.cxx

```diff
@@ -172,17 +172,18 @@ MakeClusters(const std::vector<std::string> &treeNames, const std::vector<std::s
    }
 
    // Here we "fuse" together clusters if the number of clusters is too big with respect to
-   // the number of slots, otherwise we can incurr in an overhead which is big enough
+   // the number of slots, otherwise we can incur in an overhead which is big enough
    // to make parallelisation detrimental to performance.
    // For example, this is the case when, following a merging of many small files, a file
    // contains a tree with many entries and with clusters of just a few entries each.
-   // The criterion according to which we fuse clusters together is to have at most
-   // TTreeProcessorMT::GetMaxTasksPerFilePerWorker() clusters per file per slot.
-   // For example: given 2 files and 16 workers, at most
-   // 16 * 2 * TTreeProcessorMT::GetMaxTasksPerFilePerWorker() clusters will be created, at most
-   // 16 * TTreeProcessorMT::GetMaxTasksPerFilePerWorker() per file.
+   // Another problematic case is a high number of slots (e.g. 256) coupled with a high number
+   // of large files (e.g. 1000 files): the large amount of files might result in a large amount
+   // of tasks, but the elevated concurrency level makes the little synchronization required by
+   // task initialization very expensive. In this case it's better to simply process fewer, larger tasks.
+   // The criterion according to which we fuse clusters together is to have around
+   // TTreeProcessorMT::GetTasksPerWorkerHint() clusters per file per slot.
```

Comment:
This should be `to have around TTreeProcessorMT::GetTasksPerWorkerHint() clusters per slot`, right? (removed `per file`)

Reply:
right, thanks!